src/create_discretized_sphere.cpp

```diff
@@ -0,0 +1,109 @@
+#include<ros/ros.h>
+#include "reuleaux/PointArray.h"
+#include "geometry_msgs/Point.h"
+#include<octomap/octomap.h>
+#include<octomap/MapCollection.h>
+#include <octomap/math/Utils.h>
+#include <reuleaux/sphere_discretization.h>
+using namespace octomap;
+using namespace std;
+using namespace octomath;
+using namespace sphere_discretization;
+
+
+
+int main(int argc, char **argv)
+{
+    ros::init(argc, argv, "rand_points");
+    ros::NodeHandle n;
+  
+    ros::Publisher point_pub = n.advertise<reuleaux::PointArray>("points", 1000);
+
+  //ros::Rate loop_rate(10);
+  ros::Rate poll_rate(100);
+while(point_pub.getNumSubscribers() == 0)
+    poll_rate.sleep();
+  int count = 0;
+  while (ros::ok())
+  {
+      
+     float HI=-1.5, LO=1.5;
+     unsigned char maxDepth = 16;
+     unsigned char minDepth = 0;
+    
+
+    
+
+    SphereDiscretization sd;
+    float r=1;
+    float resolution =0.1;
+    point3d origin=point3d(0,0,0);
+    OcTree* tree=sd.generateBoxTree(origin, r, resolution);
+    cout<<"Leaf"<<endl;
+    std::vector<point3d> newData;
+    for (OcTree::leaf_iterator it=tree->begin_leafs(maxDepth), end=tree->end_leafs();it !=end;++it){
+	
+	newData.push_back(it.getCoordinate());
+	
+		
+ 	}
+    
+    cout<<"The # of leaf nodes in first tree: "<<tree->getNumLeafNodes()<<endl;
```

Comment:
Replace all `cout` lines with `ROS_INFO` or `ROS_INFO_STREAM`.  This way, messages can be shown on rqt_console, will be saved to roslog, and users can set the node verbosity level to silence messages they don't want.